SU2_CFD/include/numerics/turbulent/turb_sources.hpp

```diff
@@ -725,20 +729,68 @@ class CSourcePieceWise_TurbSST final : public CNumerics {
       }
 
-      su2double pk = Eddy_Viscosity_i * pow(StrainMag, 2) - 2.0 / 3.0 * Density_i * ScalarVar_i[0] * diverg;
-      pk = max(0.0, min(pk, 20.0 * beta_star * Density_i * ScalarVar_i[1] * ScalarVar_i[0]));
+      su2double pk = Eddy_Viscosity_i * pow(P_Base, 2);
+
+      if (sstParsedOptions.version != SST_OPTIONS::MODIFIED) {
+       /* in case of unmodified, we add the divergence terms */
+        if (sstParsedOptions.version == SST_OPTIONS::V1994) {
+          /*--- INTRODUCE THE SST-V1994 BUG WHERE DIVERGENCE TERM IS MISSING ---*/
+          //pk -=  (Eddy_Viscosity*2.0/3.0*diverg*diverg + 2.0 / 3.0 * Density_i * ScalarVar_i[0] * diverg);
+          pk = pk - 2.0 / 3.0 * Density_i * ScalarVar_i[0] * diverg;
+        } else {
+         /* Note that we have to make the stress tensor tau_ij consistent everywhere when we neglect (or not) the divergence */
+         pk -=  (Eddy_Viscosity_i*(2.0 / 3.0 * diverg * diverg) + 2.0 / 3.0 * Density_i * ScalarVar_i[0] * diverg);
+        }
+      } 
+
+      //pk = pk - 2.0 / 3.0 * Density_i * ScalarVar_i[0] * diverg;
+      pk = min(pk, ProdLimConstant * beta_star * Density_i * ScalarVar_i[1] * ScalarVar_i[0]);
+      pk = max(0.0, pk);
+
+      su2double zeta;
+
+      if (sstParsedOptions.version == SST_OPTIONS::V1994){
+        /*--- no production limiter on w-equation for SST1994 ---*/
+        zeta = max(ScalarVar_i[1], VorticityMag * F2_i / a1);
+        /*--- ---*/
+        //pw = (alfa_blended * Density_i) * pw;
+      } else {
+        /*--- production limiter on w-equation for SST2003 ---*/
+        zeta = max(ScalarVar_i[1], StrainMag_i * F2_i / a1);
+        //pw = (alfa_blended * Density_i) * max(0.0, min(pw, ProdLimConstant * beta_star * Density_i * ScalarVar_i[1] * ScalarVar_i[0]));
+      }
 
-      const su2double VorticityMag = GeometryToolbox::Norm(3, Vorticity_i);
-      const su2double zeta = max(ScalarVar_i[1], VorticityMag * F2_i / a1);
-      su2double pw = alfa_blended * Density_i * max(pow(StrainMag, 2) - 2.0 / 3.0 * zeta * diverg, 0.0);
+      su2double pw = pow(StrainMag, 2); 
+
+      if (sstParsedOptions.version != SST_OPTIONS::MODIFIED) {
+       /* in case of unmodified, we add the divergence terms */
+        if (sstParsedOptions.version == SST_OPTIONS::V1994) {
+          /*--- INTRODUCE THE SST-V1994 BUG WHERE DIVERGENCE TERM IS MISSING ---*/
+          /* --- note that mu_t = rho*a1*k/zeta and we can make this term independent of k ---*/
+          // the correct production with the additional divergence term
+          // pw -=  (alfa_blended*Density_i * (2.0 / 3.0 * diverg*diverg) +  2.0 / 3.0 * zeta * diverg);
+          /*--- the tke term only ---*/
+          pw = pw - 2.0 / 3.0 * zeta * diverg;
+        } else {
+         /* Note that we have to make the stress tensor tau_ij consistent everywhere when we neglect (or not) the divergence */
+         pw = pw - (2.0 / 3.0 * diverg * diverg + 2.0 / 3.0 * zeta * diverg);
```

Comment:
Wasn't the plan to leave this out for now?

Reply:
Yes, we can comment it and make this option unavailable. However, we do need the term for v1994 at this moment since it is currently used.

Reply:
OK, I am removing everything in turb_sources that we do not need right now, and we can get back to this commit when introducing the V1994 and V2003 models where we keep all divergence terms. We can then focus on making sure that V2003 is correct, and having the new options list in place.

Reply:
Perfect.